src/coreclr/jit/gentree.cpp

```diff
@@ -25284,6 +25284,24 @@ bool GenTreeHWIntrinsic::OperIsCreateScalarUnsafe() const
     }
 }
 
+//------------------------------------------------------------------------
+// OperIsBitwiseHWIntrinsic: Is this HWIntrinsic a bitwise logic intrinsic node.
+//
+// Return Value:
+//    Whether "this" is a bitwise logic intrinsic node.
+//
+bool GenTreeHWIntrinsic::OperIsBitwiseHWIntrinsic() const
+{
+#if defined(TARGET_XARCH)
+    NamedIntrinsic intrinsicId = GetHWIntrinsicId();
+    return Compiler::gtIsBitwiseIntrinsic(intrinsicId, GT_AND) || Compiler::gtIsBitwiseIntrinsic(intrinsicId, GT_OR) ||
+           Compiler::gtIsBitwiseIntrinsic(intrinsicId, GT_XOR) ||
+           Compiler::gtIsBitwiseIntrinsic(intrinsicId, GT_AND_NOT);
+#else // !TARGET_XARCH
+    return false;
+#endif
```

Comment:
This should be fine to be general and not XARCH only.

Reply:
Thanks for reviewing, sharing the long term plan! 

Trying to confirm, if I understand it correctly, we will stick with the current implementation and continue the review process, right?